ivy.el

```diff
@@ -3524,7 +3524,9 @@ possible match.  See `all-completions' for further information."
      (lambda (x)
        (let* ((buf (get-buffer x))
               (dir (buffer-local-value 'default-directory buf))
-              (face (if (and dir (file-remote-p (abbreviate-file-name dir)))
+              (face (if (and dir
+                             (ignore-errors
```

Comment:
You should almost never use `ignore-errors` because it

1. swallows other potentially important errors;
2. pretty bad at documenting the intent and actual reason for it being there.

In case of TRAMP-related errors, it's better to explicitly disable TRAMP handlers by let-binding `tramp-mode` to `nil`.  In fact, here is a gem from my configuration of `ivy` since the beginning:

```
  (defun init-ivy-switch-buffer
      (function &rest ...)
    (let (tramp-mode)
      (apply function ...)))
  (advice-add #'ivy-switch-buffer
              :around
              #'init-ivy-switch-buffer)
```

Reply:
>     1. swallows other potentially important errors;

In this case it doesn't though. It simply returns `nil` if `file-remote-p` could not return a truthy value. That's exactly what we want from `file-remote-p`.